enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,266 @@
+
+- When creating the hosted cluster, we need to override the `--release-image` option to specify the manifest listed release image in order to be able to deploy on a worker node of any architecture. Optionally, the config could provide a flag to indicate that it is being installed on a heterogeneous architecture cluster in which case
+it will pull down a manifest listed release payload rather than the x86 release payload.
+- For the hosted control plane to work on non x86 architecture nodes, the following image references would need changing:
+  - the release image fixtures referencing x86 imagestreams. these would need to be changed to manifestlist references or there would need to be imagestreams added for individual architectures and they would be picked based on the node architecture.
+  - references to some images like cluster-api-provider-aws, apiserver-network-proxy etc.. are x86. these would need to change to provide the images for other architectures too. these images can be overridden through annotations today.
+- Another option would be to mandate an x86 worker node be present for Hypershift and nodeselect the x86 node for the hosted cluster control plane deployment.
+- When creating the nodepool, the defaults for instances types and AMIs are x86 specific. The instance types can be overridden but not the AMIs. An option needs to be added to the create cluster command to override AMIs.
+
+### Console
+
+As of today console filters content based on control plane architecture. This needs to be entirely removed or made configurable for heterogeneous clusters.
+
+### Install
+
+For Phase 1, the way to add more workers to the cluster is to handcraft a machineset manifest file for a worker node and specify the architecture specific details like instance type, image references and deploy the machineset to get a worker node. This can be pseudo automated further by creating the manifests using `openshift-install create manifests` and modifying
+or adding worker manifest files. Eventually the installer should support being able to declare differing compute and control plane sections in the install-config mainly allowing architectures to be different. In doing so it would also generate appropriate machineset and machineconfig files tailored to the specific architecture. The machineset file would contain
+the instance types and region corresponding to the architecture chosen for the workers.
+
+### Open Questions
+
+- What would be the method to provide recommendation updates when moving from a homogeneous graph to a heterogeneous graph ?
+- Do OLM operator images have to include all architectures, even ones that are not supported by the operator ?
```

Comment:
No.  If the publisher of an olm operator does not want to support a particular architecture, that is up to them.

(note that i'd expect all *redhat* operators to support all architectures that OCP supports, however).

This gets into the content filtering the console needs to do.  If i have a cluster that has ARM and x86 nodes:

* it's ok to install an operator that only supports ARM (there are workers where it can run)
* it's ok to install an operator that only supports x86 (there are workers where it can run)
* it's not ok to install an operator that only supports ppc + s390 (there will be nowhere it can run)
* sharp edge: it's not ok to install an operator that only supports ARM but expects to deploy a daemonset that runs on all workers.  I don't know that this scenario is likely enough to worry about for now.

So the console will need to be able to appropriately filter the set of olm operators that, if installed, will be able to find a worker to run on(meaning there is some overlap between the set of architectures the worker nodes are running on, and the set of architectures the olm operator claims to support)

Reply:
makes sense. updated the console section to reflect this and removed the question.